client.go

```diff
@@ -41,48 +41,56 @@ type SingleServerClient struct {
 	isLocalTest bool
 }
 
-func NewClientImpl(deployFile string, targetName string, isLocalTest bool) (*SingleServerClient, error) {
+// Client which communicates with multiple underlying servers at once. Used if
+// the target is actually a group of servers
+type MultiServerClient struct {
+	app     Application
+	appDir  string
+	clients []Client
+}
+
+func NewClient(deployFile string, targetName TargetName, isLocalTest bool) (*MultiServerClient, error) {
 	app, err := ApplicationFromConfig(true, deployFile)
 	if err != nil {
 		return nil, err
 	}
 
-	target := app.Target(targetName)
-	if target == nil {
-		return nil, fmt.Errorf("Invalid target: '%s'", targetName)
-	}
+	// Create all SingleServerClients
+	clients := []Client{}
+	for _, target := range app.Targets(targetName) {
+		localPort := target.Base
```

Comment:
Each 'Target' is a new machine, so we don't get port clashes.

It doesn't need to be the same as the base, however, clients do need to know what port camus is running on. I'm guessing using the 'base' property for 2 things (the port camus listens on, and the port camus starts provisioning from) makes life easier in the early days?

Regardless - Not entirely clear as to what changes you are proposing in this particular case. If you would like to continue the discussion, may I suggest moving this over to a github issue.